### PR TITLE
Variable assignment benchmarks

### PR DESCRIPTION
**Pull request by DanNixon**

### CODE REVIEW

File: python/tests/benchmark_variable.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (SimonHeybrock) [inline python/tests/benchmark_variable.py]: Any reason this is in the `tests` folder? If not, move it to a new folder `benchmark`?

File: python/tests/benchmark_variable.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# @file
+import time
+import scipp as sc
+from scipp import Dim
+import numpy as np
+
+
+def assign_from_numpy_1d(size):
+    a = np.ndarray(dtype=np.float64, shape=(size, ))
+    a.fill(0)
```
Comment (SimonHeybrock) [inline python/tests/benchmark_variable.py]: Why not use `np.zeros()`?

Comment (DanNixon) [reply python/tests/benchmark_variable.py]: Interesting. Using `zeros` gives the same oddly high performance I saw earlier.

Comment (SimonHeybrock) [reply python/tests/benchmark_variable.py]: 😨

Comment (SimonHeybrock) [reply python/tests/benchmark_variable.py]: What about `ones` or `arange`?

Comment (DanNixon) [reply python/tests/benchmark_variable.py]: `ones` is the same performance as the create then fill I had above.

File: benchmark/variable_benchmark.cpp [hunk 3]
```diff
@@ -143,4 +143,28 @@ static void BM_Variable_binary_with_VariableProxy(benchmark::State &state) {
 BENCHMARK(BM_Variable_binary_with_Variable);
 BENCHMARK(BM_Variable_binary_with_VariableProxy);
 
+static void BM_Variable_assign_1d(benchmark::State &state) {
+  const auto size = state.range(0);
+
+  const auto a = createVariable<double>(Dims{Dim::X}, Shape{size});
+  auto b = createVariable<double>(Dims{Dim::X}, Shape{size});
```
Comment (igudich) [inline benchmark/variable_benchmark.cpp]: createVariable -> makeVariable

Comment (DanNixon) [reply benchmark/variable_benchmark.cpp]: I thought `createVariable` was the correct function to use now? Has this changed again?

Comment (DanNixon) [reply benchmark/variable_benchmark.cpp]: Oh. Apparently it had.

**Pull request merged**